tree/treeplayer/inc/ROOT/TCsvDS.hxx

```diff
@@ -0,0 +1,54 @@
+#ifndef ROOT_TCSVTDS
+#define ROOT_TCSVTDS
+
+#include "ROOT/TDataSource.hxx"
+
+#include <map>
+#include <vector>
+#include <TRegexp.h>
+
+namespace ROOT {
+namespace Experimental {
+namespace TDF {
+
+class TCsvDS final : public ROOT::Experimental::TDF::TDataSource {
+
+private:
+   typedef std::vector<void *> Record;
+
+   unsigned int fNSlots = 0U;
+   std::string fFileName;
+   char fDelimiter;
```

Comment:
I am not an expert of csvs. Is it true that the separator can be only a single character?

Reply:
well, [rules as written](https://tools.ietf.org/html/rfc4180) there is only one true delimiter: comma. It's a reasonable extension to allow arbitrary characters as delimiters (and any number of white spaces in between!) but at least as a first approximation I'm not sure we need more than that

Reply:
I think you both have a point, it is just a matter of deciding what we would like to support. I believe comma plus eventually some other character is enough for now, although extending it would not be a problem.

Reply:
I support this pragmatic approach.